tests/Unit/PointwiseFunctions/AnalyticSolutions/GeneralRelativity/Test_WrapGh.cpp

```diff
@@ -0,0 +1,266 @@
+          tmpl::list<Tags::deriv<
+              gr::Tags::SpatialMetric<wrapped_solution.volume_dim,
+                                      Frame::Inertial, DataVector>,
+              tmpl::size_t<wrapped_solution.volume_dim>, Frame::Inertial>>{}));
+  const auto wrapped_inv_g = get<gr::Tags::InverseSpatialMetric<
+      wrapped_solution.volume_dim, Frame::Inertial, DataVector>>(
+      wrapped_solution.variables(
+          x, t,
+          tmpl::list<gr::Tags::InverseSpatialMetric<
+              wrapped_solution.volume_dim, Frame::Inertial, DataVector>>{}));
+  const auto wrapped_extrinsic_curvature = get<gr::Tags::ExtrinsicCurvature<
+      wrapped_solution.volume_dim, Frame::Inertial, DataVector>>(
+      wrapped_solution.variables(
+          x, t,
+          tmpl::list<gr::Tags::ExtrinsicCurvature<
+              wrapped_solution.volume_dim, Frame::Inertial, DataVector>>{}));
+  const auto wrapped_sqrt_det_g =
+      get<gr::Tags::SqrtDetSpatialMetric<DataVector>>(
+          wrapped_solution.variables(
+              x, t, tmpl::list<gr::Tags::SqrtDetSpatialMetric<DataVector>>{}));
+
+  // Check that the solution and the wrapped solution returned the same
+  // results
+  CHECK(lapse == wrapped_lapse);
+  CHECK(dt_lapse == wrapped_dt_lapse);
+  CHECK(d_lapse == wrapped_d_lapse);
+
+  CHECK(shift == wrapped_shift);
+  CHECK(dt_shift == wrapped_dt_shift);
+  CHECK(d_shift == wrapped_d_shift);
+
+  CHECK(g == wrapped_g);
+  CHECK(dt_g == wrapped_dt_g);
+  CHECK(d_g == wrapped_d_g);
+
+  // Check that the wrapped solution returns the correct psi, pi, phi
+  const auto psi = gr::spacetime_metric(lapse, shift, g);
+  const auto phi =
+      GeneralizedHarmonic::phi(lapse, d_lapse, shift, d_shift, g, d_g);
+  const auto pi =
+      GeneralizedHarmonic::pi(lapse, dt_lapse, shift, dt_shift, g, dt_g, phi);
+
+  const auto wrapped_gh_vars = wrapped_solution.variables(
+      x, t,
+      tmpl::list<gr::Tags::SpacetimeMetric<wrapped_solution.volume_dim,
+                                           Frame::Inertial, DataVector>,
+                 GeneralizedHarmonic::Tags::Pi<wrapped_solution.volume_dim,
+                                               Frame::Inertial>,
+                 GeneralizedHarmonic::Tags::Phi<wrapped_solution.volume_dim,
+                                                Frame::Inertial>>{});
+  CHECK(psi == get<gr::Tags::SpacetimeMetric<wrapped_solution.volume_dim,
+                                             Frame::Inertial, DataVector>>(
+                   wrapped_gh_vars));
+  CHECK(pi ==
+        get<GeneralizedHarmonic::Tags::Pi<wrapped_solution.volume_dim,
+                                          Frame::Inertial>>(wrapped_gh_vars));
+  CHECK(phi ==
+        get<GeneralizedHarmonic::Tags::Phi<wrapped_solution.volume_dim,
+                                           Frame::Inertial>>(wrapped_gh_vars));
+
+  test_serialization(wrapped_solution);
+  // test operator !=
+  CHECK_FALSE(wrapped_solution != wrapped_solution);
```

Comment:
I think the test for `operator !=` and `==` should be more rigorous . Maybe construct two KerrSchild solutions with different masses/spins and make sure they're different?